src/geometry/manifold/instances/units_of_normed_algebra.lean

```diff
@@ -62,4 +63,76 @@ variables {𝕜 : Type*} [nontrivially_normed_field 𝕜] [normed_algebra 𝕜 R
 instance : smooth_manifold_with_corners 𝓘(𝕜, R) Rˣ :=
 open_embedding_coe.singleton_smooth_manifold_with_corners 𝓘(𝕜, R)
 
+lemma smooth_mul :
+  smooth (𝓘(𝕜, R).prod 𝓘(𝕜, R)) 𝓘(𝕜, R) (λ (p : Rˣ × Rˣ), p.fst * p.snd) :=
+begin
+  apply cont_mdiff.smooth,
+  rw cont_mdiff_iff,
+  split,
+  { continuity },
+  { intros x' y',
+    simp,
```

Comment:
Note that this is a non-terminal simp, and so will need to be replaced with `simp only` or similar ("squeezed"). You might be able to do `squeeze_simp` to work out the correct invocation.